app/coffee/Features/Collaborators/CollaboratorsInviteHandler.coffee

```diff
@@ -0,0 +1,118 @@
+Project = require("../../models/Project").Project
+ProjectInvite = require("../../models/ProjectInvite").ProjectInvite
+mimelib = require("mimelib")
+logger = require('logger-sharelatex')
+ContactManager = require "../Contacts/ContactManager"
+CollaboratorsEmailHandler = require "./CollaboratorsEmailHandler"
+Async = require "async"
+PrivilegeLevels = require "../Authorization/PrivilegeLevels"
+Errors = require "../Errors/Errors"
+Crypto = require 'crypto'
+
+module.exports = CollaboratorsInviteHandler =
+
+	getAllInvites: (projectId, callback=(err, invites)->) ->
+		logger.log {projectId}, "fetching invites from mongo"
+		ProjectInvite.find {projectId: projectId}, (err, invites) ->
+			if err?
+				logger.err {err, projectId}, "error getting invites from mongo"
+				return callback(err)
+			callback(null, invites)
+
+	inviteToProject: (projectId, sendingUserId, email, privileges, callback=(err,invite)->) ->
+		logger.log {projectId, sendingUserId, email, privileges}, "adding invite"
+		Crypto.randomBytes 24, (err, buffer) ->
+			if err?
+				logger.err {err, projectId, sendingUserId, email}, "error generating random token"
+				return callback(err)
+			token = buffer.toString('hex')
+			invite = new ProjectInvite {
+				email: email
+				token: token
+				sendingUserId: sendingUserId
+				projectId: projectId
+				privileges: privileges
+			}
+			invite.save (err, invite) ->
+				if err?
+					logger.err {err, projectId, sendingUserId, email}, "error saving token"
+					return callback(err)
+				CollaboratorsEmailHandler.notifyUserOfProjectInvite projectId, email, invite
+				callback(null, invite)
+
+	revokeInvite: (projectId, inviteId, callback=(err)->) ->
+		logger.log {projectId, inviteId}, "removing invite"
+		ProjectInvite.remove {projectId: projectId, _id: inviteId}, (err) ->
+			if err?
+				logger.err {err, projectId, inviteId}, "error removing invite"
+				return callback(err)
+			callback(null)
+
+	getInviteByToken: (projectId, tokenString, callback=(err,invite)->) ->
+		logger.log {projectId, tokenString}, "fetching invite by token"
+		ProjectInvite.findOne {projectId: projectId, token: tokenString}, (err, invite) ->
+			if err?
+				logger.err {err, projectId}, "error fetching invite"
+				return callback(err)
+			if !invite
```

Comment:
!invite? would be a bit better on the case that false or similar was passed through somehow.